### PR TITLE
docs(ci): support debugging on CI server through ssh

### LINKED ISSUE #717: feat(ci): support debugging on CI server through ssh
### Description
As a contributor I want to be able to debug the failing tests on the CI server directly so that I can figure out what is going wrong even if my tests are only failing on the CI server and not on my local machine (which is a very common scenario at present...)

Use this (?) => https://github.com/marketplace/actions/debugging-with-ssh

### Acceptance Criteria
1. Works out of the box with VSCode debugger (preferably, but other is acceptable too)
2. GIF tutorial in CONTRIBUTING.md showing exactly how it's done in 30 seconds or less.

cc: @takeutak @sfuji822 @hartm @jonathan-m-hamilton @AzaharaC @jordigiam @kikoncuo @jagpreetsinghsasan

### PR DESCRIPTION
**Pull request by Leeyoungone**
Fixes #717

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -407,6 +411,18 @@ Breakpoints will work as long as you are debugging code in the same package.
...
+
+#### Getting into the SSH connection
+  Upload your public key onto github if not done so already. A public key is necessary to join the ssh connection to use upterm. A comprehensive guide is linked here (insert link here).
+
+  Edit the `ci.yml` file by following the steps here (insert link) to create a upterm session within `.github/workflows` by adding a new build step. 
+  
+  By creating a PR for the edited `ci.yml` file, this will the CI to run their tests. There are two ways to navigate to CIs. 
+    1) Go to the PR and click the `checks` tab
+    2) Go to the `Actions` tab within the main Hyperledger Cactus Repository
+
```
Comment (petermetz) [inline CONTRIBUTING.md:424]:
```sh
ssh **********:***********@uptermd.upterm.dev
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -407,6 +411,18 @@ Breakpoints will work as long as you are debugging code in the same package.
 
 > Source map support is partial at this point but actively being worked on.
 
+### Debugging the CI server through the ssh
+
+#### Getting into the SSH connection
+  Upload your public key onto github if not done so already. A public key is necessary to join the ssh connection to use upterm. A comprehensive guide is linked here (insert link here).
+
+  Edit the `ci.yml` file by following the steps here (insert link) to create a upterm session within `.github/workflows` by adding a new build step.
```
Comment (petermetz) [inline CONTRIBUTING.md:419]:
```yaml
    - name: Setup upterm session
      uses: lhotari/action-upterm@v1
      with: 
        repo-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment (petermetz) [reply CONTRIBUTING.md:419]: Make sure to mention that the SSH upterm session has to come after the checkout step:
- uses: actions/checkout@v2.3.4

**Review by petermetz — CHANGES_REQUESTED** [review]
Added comments inline

File: CONTRIBUTING.md (hunk 1 above)
Comment (Leeyoungone) [reply CONTRIBUTING.md]: neat! thank you!

File: CONTRIBUTING.md (hunk 2 above)
Comment (Leeyoungone) [reply CONTRIBUTING.md]: got it! tytyvm :)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -29,4 +29,4 @@ jobs:
 
     - uses: actions/checkout@v2.3.4
 
-    - run: ./tools/ci.sh
+    - run: ./tools/ci.sh
```
Comment (petermetz) [inline .github/workflows/ci.yml:32]: @Leeyoungone This might be because of a new line was either removed or added. Please if possible undo the change so that the PR is only changing the docs (I'm nit picking, apologies)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -99,21 +99,23 @@ Further reading:
...
     2. Be aware that we are using git commit hooks for the automation of certain mundane tasks such as applying the required code style and formatting so your code will be wrapped at 80 characters each line automatically. If you wish to see how your changes will be altered by the formatter you can run the `npm run prettier` command from a terminal or install an IDE extension for the `Prettier` tool that can do the same (VSCode has one that is known to work).
 8. Ensure your branch is rebased onto the `upstream` main branch where `upstream` is fancy git talk for the main Cactus repo on Github (the one you created your fork from).
-   1. If you are having trouble, there are many great resources out there (so we will not write another here).
+   1. **Do not** duplicate your pull request after it has been reviewed. Duplication here means closing the existing PR and then opening a brand new one which does not contain the review history anymore. If you encounter issues with version control that you do not know how to solve the maintainers will be happy to assist to ensure that you do not need to open a new pull request from scratch.
```
Comment (petermetz) [inline CONTRIBUTING.md:102]: @Leeyoungone Please double check if these are changes leftover from a rebase/conflict resolution gone sideways. If you know for a fact that there's no changes in this file anymore that you wanted to make in this PR (because all of it is now in the BUILD.md file) then you can also just restore CONTRIBUTING.md to whatever state it is currently in on the main branch.

**Review by petermetz — CHANGES_REQUESTED** [review]

**Comment by Leeyoungone** [comment]
@petermetz Thank you! I think I got it now :)

**Review by petermetz — APPROVED** [review]

**Review by takeutak — APPROVED** [review]
